src/uikit.cs

```diff
@@ -21315,4 +21340,103 @@ interface UITableViewDiffableDataSource<SectionIdentifierType,ItemIdentifierType
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface UIActivityItemsConfiguration : UIActivityItemsConfigurationReading {
+
+		[NullAllowed, Export ("localObject", ArgumentSemantic.Strong)]
+		NSObject LocalObject { get; set; }
+
+		[Export ("supportedInteractions", ArgumentSemantic.Copy)]
+		NSString [] WeakSupportedInteractions { get; set; }
+
+		[Advice ("Uses 'UIActivityItemsConfigurationMetadataKey' constants in the function handler.")]
+		[NullAllowed, Export ("metadataProvider", ArgumentSemantic.Strong)]
+		UIActivityItemsConfigurationMetadataProviderHandler MetadataProvider { get; set; }
+
+		[Advice ("Uses 'UIActivityItemsConfigurationMetadataKey' constants in the function handler.")]
+		[NullAllowed, Export ("perItemMetadataProvider", ArgumentSemantic.Strong)]
+		UIActivityItemsConfigurationPerItemMetadataProviderHandler PerItemMetadataProvider { get; set; }
+
+		[Advice ("Uses 'UIActivityItemsConfigurationPreviewIntent' enum constants in the function handler.")]
+		[NullAllowed, Export ("previewProvider", ArgumentSemantic.Strong)]
+		UIActivityItemsConfigurationPreviewProviderHandler PreviewProvider { get; set; }
+
+		[NullAllowed, Export ("applicationActivitiesProvider", ArgumentSemantic.Strong)]
+		Func<UIActivity []> ApplicationActivitiesProvider { get; set; }
+
+		[Static]
+		[Export ("activityItemsConfigurationWithObjects:")]
+		UIActivityItemsConfiguration Create (INSItemProviderWriting [] objects);
+
+		[Static]
+		[Export ("activityItemsConfigurationWithItemProviders:")]
+		UIActivityItemsConfiguration Create (NSItemProvider [] itemProviders);
+
+		[Export ("initWithObjects:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (INSItemProviderWriting[] objects);
+
+		[Export ("initWithItemProviders:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (NSItemProvider[] itemProviders);
+	}
+
+	interface IUIActivityItemsConfigurationReading { }
+
+	[NoWatch, NoTV, iOS (13,0)]
+	[Protocol]
+	interface UIActivityItemsConfigurationReading {
+
+		[Abstract]
+		[Export ("itemProvidersForActivityItemsConfiguration", ArgumentSemantic.Copy)]
+		NSItemProvider [] ItemProvidersForActivityItemsConfiguration { get; }
+
+		[Export ("activityItemsConfigurationSupportsInteraction:")]
+		bool GetActivityItemsConfigurationSupportsInteraction (NSString activityItemsConfigurationInteraction);
+
+		[Export ("activityItemsConfigurationMetadataForKey:")]
+		[return: NullAllowed]
+		NSObject GetActivityItemsConfigurationMetadata (NSString activityItemsConfigurationMetadataKey);
+
+		[Export ("activityItemsConfigurationMetadataForItemAtIndex:key:")]
+		[return: NullAllowed]
+		NSObject GetActivityItemsConfigurationMetadata (nint index, NSString activityItemsConfigurationMetadataKey);
+
+		[Export ("activityItemsConfigurationPreviewForItemAtIndex:intent:suggestedSize:")]
+		[return: NullAllowed]
+		NSItemProvider GetActivityItemsConfigurationPreview (nint index, NSString activityItemsConfigurationPreviewIntent, CGSize suggestedSize);
+
```

Comment:
In all the methods, or most of them we have `GetActivityItemsConfiguration` then either preview or metadata. Can we just use GetMetadata and GetPreview?

Reply:
You can't because this is a protocol :) Also I think the name is important because this is a protocol and can be implemented by several, also I think this avoids a round trip to documentation

Reply:
and also goes by hand with the property `ItemProvidersForActivityItemsConfiguration`

Reply:
Meh.. I'm going to accept it because Swift uses the same.. but what a horrible name.

Reply:
And swift uses the same

```
optional func activityItemsConfigurationPreviewForItem(at index: Int, 
                                                intent: UIActivityItemsConfigurationPreviewIntent, 
                                         suggestedSize: CGSize) -> NSItemProvider?
```